core/src/main/java/com/linecorp/armeria/client/Http2ResponseDecoder.java

```diff
@@ -53,6 +53,7 @@
 
     private final Http2Connection conn;
     private final Http2ConnectionEncoder encoder;
+    private volatile long goAwayErrorCode = -1; // -1 if not received GOAWAY yet
```

Comment:
Since this is a single connection, won't this always be read / write from the same thread?

Reply:
Fixed